### PR TITLE
Added StatusesGet method

### PR DESCRIPTION
**Pull request by abhitrueprogrammer**
Added the StatusesGet() method. Returns a array of objects with id and description on successful execution. Gets the status of submissions  using judge0

### CODE REVIEW

File: info.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	}
+
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("error reading response body: %v", err)
+	
+	}
+	return body, nil
+}
```
Comment (RustyDev24) [inline info.go:37]: abstract out into a new file

File: info.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+
+func (client * Client) StatusesGet() (Statuses, error)   {
```
Comment (RustyDev24) [inline info.go:40]: Rename to `GetStatus`

File: info.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	judge0Url, err := url.Parse(client.authProvider.GetBaseURL() + "/statuses")
+	if err != nil {
+		return nil, err
+	}
+	body, err := Judge0Request(judge0Url.String(), "GET")
```
Comment (RustyDev24) [inline info.go:45]: Use `http.MethodGet`  constant

File: info.go (hunk 1 above)
Comment (RustyDev24) [reply info.go:37]: also take whatever body you want to pass into the reques

File: helpers.go [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package judge0
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+)
+func Judge0Request(url string, method string, payload []byte) ([] byte, error) {
```
Comment (RustyDev24) [inline helpers.go:9]: rename to `doRequest or smth`

File: helpers.go [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+	judgereq, err := http.NewRequest(method, url, bytes.NewReader(payload))
+	if err != nil {
+		return nil, fmt.Errorf("error creating the HTTP request: %v", err)
+	}
+	judgereq.Header.Add("Content-Type", "application/json")
+
+	resp, err := http.DefaultClient.Do(judgereq)
```
Comment (RustyDev24) [inline helpers.go:16]: add a client to the `Client` struct, use that to perform the requests

File: helpers.go [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+	if err != nil {
+		return nil, fmt.Errorf("error sending request to Judge0: %v", err)
+	}
+
```
Comment (RustyDev24) [inline helpers.go:20]: Add a check for status codes because it doesn't error then

File: info.go (hunk 2 above)
Comment (abhitrueprogrammer) [reply info.go]: The function does give out Statuses of multiple files so imo GetStatuses is more appropriate, still renamed for now

**Pull request merged**